File: 3/code.c
int lengthOfLongestSubstring(char* s) {
    int i = 0, rm = 0, len = 0, tmp;
    int hash[128] = {0};
    char c;
    while (*s != '\0') {
        c = *s;
        if (hash[c] > 0 && hash[c] >= rm) {
            rm += hash[c] - rm;
        }
        hash[c] = i + 1;
        tmp = i + 1 - rm;
        len = len >= tmp ? len : tmp;
        i++;
        s++;
    }
    return len;
}

File: 3/code.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        i = 0
        rm = 0
        leng = 0
        slen = len(s)
        hash = [0] * 128
        for i in range(0, slen):
            c = ord(s[i])
            if hash[c] > rm:
                rm += hash[c] - rm
            hash[c] = i + 1
            tmp = i + 1 -rm
            leng = max(leng, tmp)
        return leng